backup: re-enable drpc randomly in backup tests

suj-krishnan opened the pull request:
backup: re-enable drpc randomly in backup tests

Some failing backup tests were disabled earlier due to the drpc metadata issue described here - #157188. This PR re-enables the failing tests. 
Note: Only one of the backup tests `TestBackupRestoreResolveDestination` is still disabled as Extended CI fails when this test is enabled. The issue will be investigated separately.

Epic: CRDB-48935
Fixes: #157206
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/159141)
<!-- Reviewable:end -->

suj-krishnan committed:
drpc: bump drpc version to include latest changes

The following changes from drpc will be picked up with this version bump:
- Change to send grpc metadata via drpc
- Change to set soft cancel to false by default

Epic: None
Release note: None

suj-krishnan committed:
fixup! drpc: bump drpc version to include latest changes

drpc: bump drpc to include latest changes

suj-krishnan committed:
Merge branch 'master' into bump_drpc_version

suj-krishnan committed:
backup: re-enable drpc randomly in backup tests

Some failing backup tests were disabled earlier due to the drpc
metadata issue described here - #157188. This PR updates this
repo to pickup the latest drpc fixes to send and receive grpc
metadata and re-enables the failing tests.

Epic: CRDB-48935
Fixes: #157206
Release note: None
